iothub/device/src/DirectMethod/EdgeModuleDirectMethodRequest.cs

```diff
@@ -1,13 +1,18 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using System;
+using Newtonsoft.Json;
+
 namespace Microsoft.Azure.Devices.Client
 {
     /// <summary>
     /// Parameters to execute a direct method on an edge device or an edge module by an <see cref="IotHubModuleClient"/>.
     /// </summary>
-    public class EdgeModuleDirectMethodRequest : DirectMethodRequest
```

Comment:
These classes seem sufficiently different so justify separating them out.
- DirectMethodRequest is always received by the client. This can be the result of a method invocation request from service, or from an edge module
- EdgeModuleDirectMethodRequest is always sent by an edge module client. This is never received by the client.

Since one is an outgoing type and another an incoming type, they inherently require different methods.
The outgoing type is required to be able to set the payload as either an object (to be serialized by the convention during _send_. It does not require TryGet methods for fetching the payload value since the payload is set by the user themselves.
The incoming type requires the payload to be set as raw bytes since that is what we get from the transport layer. It requires TryGet methods so that the payload can be deserialized according to the convention.

Thoughts?